src/tests/parser_test.py

```diff
@@ -485,6 +485,12 @@ def test_module_publicity():
     module = parser.parse(code, "__filepath")
     assert module.is_public
 
+    module = parser.parse(code, "_directory/filepath")
+    assert module.is_public
```

Comment:
See my previous comment, this should be _private_.
Also, please add some more complex tests with sub-packages and specifically the example from the original bug (a private module in a public package).